### PR TITLE
Adding coffemaker csharp version

### PR DESCRIPTION
**Pull request by asergaz**
To run it, copy to an empty folder the following files:
- CoffeeMaker.cs
- CoffeeMaker-csharp.csproj
- Parameters.cs
- Program.cs

Then `dotnet run -d [DeviceId] -i [DpsIdScope] - k [DeviceSymmetricKey]`

### CODE REVIEW

**Comment by asergaz** [comment]
Requesting your review @dominicbetts

File: monitor-manage-device-with-iotcentral/Parameters.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+{
+    /// <summary>
+    /// Command line parameters for the SimulatedDevice sample
+    /// </summary>
+    internal class Parameters
+    {
+        [Option(
+            'i',
+            "DpsIdScope",
+            HelpText = "(Required if DeviceSecurityType is 'dps'). \nThe DPS ID Scope to use during device provisioning." +
```
Comment (dominicbetts) [inline monitor-manage-device-with-iotcentral/Parameters.cs]: For IoT Central - dps is the only security type. You could reword this here and below. You don't even need to mention DPS. In the IoT Central UI, this value is simply "ID scope"

File: monitor-manage-device-with-iotcentral/Parameters.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            "\nDefaults to environment variable 'IOTHUB_DEVICE_DPS_ID_SCOPE'.")]
+        public string DpsIdScope { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_DPS_ID_SCOPE");
+
+        [Option(
+            'd',
+            "DeviceId",
+            HelpText = "(Required if DeviceSecurityType is 'dps'). \nThe device registration Id to use during device provisioning." +
```
Comment (dominicbetts) [inline monitor-manage-device-with-iotcentral/Parameters.cs]: IoT Central doesn't use "registration Id" as a term - the UI says "device ID".

File: monitor-manage-device-with-iotcentral/Parameters.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            "\nDefaults to environment variable 'IOTHUB_DEVICE_DPS_DEVICE_ID'.")]
+        public string DeviceId { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_DPS_DEVICE_ID");
+
+        [Option(
+            'k',
+            "DeviceSymmetricKey",
+            HelpText = "(Required if DeviceSecurityType is 'dps'). \nThe device symmetric key to use during device provisioning." +
```
Comment (dominicbetts) [inline monitor-manage-device-with-iotcentral/Parameters.cs]: The IoT Central terminology here is "Primary key".

**Review by dominicbetts — COMMENTED** [review]
Just a couple of minor suggestions in the code.

More generally, I'd ask why use telemetry rather than read-only properties for "Brewing", "Cup Detected", and "Device Warranty Expired" - but that choice was made in the original version.

I don't think that you need to get a review from the product team for this sample - the C# looks clear to me as sample code to illustrate device client features.

File: monitor-manage-device-with-iotcentral/Parameters.cs (hunk 1 above)
Comment (asergaz) [reply monitor-manage-device-with-iotcentral/Parameters.cs]: Changed

File: monitor-manage-device-with-iotcentral/Parameters.cs (hunk 3 above)
Comment (asergaz) [reply monitor-manage-device-with-iotcentral/Parameters.cs]: Changed

File: monitor-manage-device-with-iotcentral/Parameters.cs (hunk 2 above)
Comment (asergaz) [reply monitor-manage-device-with-iotcentral/Parameters.cs]: Done

**Pull request merged**